# Exercício de Casa 🏠 

Utilizar a tabela de dados do clima de seu estado, manipule os dados de acordo com as instruções abaixo:
    
    - calcular a média da temperatura da amostra

    - retirar nulos da coluna 'RADIACAO GLOBAL (Kj/m2)'

    - copiar o dataframe reduzindo para 3 colunas (a sua escolha) e 1000 linhas (aleatórias)

    - Bônus: normalizar coluna (qualquer uma)
    
    - Bônus II: pesquisar sobre outras formas de processamento de dados além das vistas em sala de aula

In [1]:
import pandas as pd

df = pd.read_csv('paraiba.CSV', delimiter=';', skiprows=8, encoding='latin1')

In [2]:
df.head()

Data  Hora UTC PRECIPITAÇÃO TOTAL, HORÁRIO (mm)  \
0  2023/01/01  0000 UTC                                0   
1  2023/01/01  0100 UTC                              NaN   
2  2023/01/01  0200 UTC                             10,4   
3  2023/01/01  0300 UTC                               ,6   
4  2023/01/01  0400 UTC                                0   

  PRESSAO ATMOSFERICA AO NIVEL DA ESTACAO, HORARIA (mB)  \
0                                             1009,8      
1                                                NaN      
2                                             1009,3      
3                                             1008,2      
4                                             1007,4      

  PRESSÃO ATMOSFERICA MAX.NA HORA ANT. (AUT) (mB)  \
0                                          1009,9   
1                                             NaN   
2                                          1009,8   
3                                          1009,3   
4                                          1008,2   

  PRESSÃO ATMOSFERICA MIN. NA HORA ANT. (AUT) (mB) RADIACAO GLOBAL (Kj/m²)  \
0                                           1009,4                     NaN   
1                                              NaN                     NaN   
2                                           1009,3                     NaN   
3                                           1008,2                     NaN   
4                                           1007,4                     NaN   

  TEMPERATURA DO AR - BULBO SECO, HORARIA (°C)  \
0                                           24   
1                                          NaN   
2                                         23,3   
3                                         22,7   
4                                         22,7   

  TEMPERATURA DO PONTO DE ORVALHO (°C)  \
0                                 22,8   
1                                  NaN   
2                                 22,3   
3                                 21,8   
4                                 21,9   

  TEMPERATURA MÁXIMA NA HORA ANT. (AUT) (°C)  \
0                                       24,2   
1                                        NaN   
2                                       24,3   
3                                       23,3   
4                                       22,7   

  TEMPERATURA MÍNIMA NA HORA ANT. (AUT) (°C)  \
0                                       23,9   
1                                        NaN   
2                                       23,3   
3                                       22,6   
4                                       22,5   

  TEMPERATURA ORVALHO MAX. NA HORA ANT. (AUT) (°C)  \
0                                               23   
1                                              NaN   
2                                             23,1   
3                                             22,2   
4                                               22   

  TEMPERATURA ORVALHO MIN. NA HORA ANT. (AUT) (°C)  \
0                                             22,7   
1                                              NaN   
2                                             22,3   
3                                             21,5   
4                                             21,6   

   UMIDADE REL. MAX. NA HORA ANT. (AUT) (%)  \
0                                      93.0   
1                                       NaN   
2                                      94.0   
3                                      95.0   
4                                      96.0   

   UMIDADE REL. MIN. NA HORA ANT. (AUT) (%)  \
0                                      93.0   
1                                       NaN   
2                                      92.0   
3                                      93.0   
4                                      95.0   

   UMIDADE RELATIVA DO AR, HORARIA (%)  VENTO, DIREÇÃO HORARIA (gr) (° (gr))  \
0                                 93.0                                 160.0 

In [7]:
df.dtypes

Data                                                      object
Hora UTC                                                  object
PRECIPITAÇÃO TOTAL, HORÁRIO (mm)                          object
PRESSAO ATMOSFERICA AO NIVEL DA ESTACAO, HORARIA (mB)     object
PRESSÃO ATMOSFERICA MAX.NA HORA ANT. (AUT) (mB)           object
PRESSÃO ATMOSFERICA MIN. NA HORA ANT. (AUT) (mB)          object
RADIACAO GLOBAL (Kj/m²)                                   object
TEMPERATURA DO AR - BULBO SECO, HORARIA (°C)              object
TEMPERATURA DO PONTO DE ORVALHO (°C)                      object
TEMPERATURA MÁXIMA NA HORA ANT. (AUT) (°C)                object
TEMPERATURA MÍNIMA NA HORA ANT. (AUT) (°C)                object
TEMPERATURA ORVALHO MAX. NA HORA ANT. (AUT) (°C)          object
TEMPERATURA ORVALHO MIN. NA HORA ANT. (AUT) (°C)          object
UMIDADE REL. MAX. NA HORA ANT. (AUT) (%)                 float64
UMIDADE REL. MIN. NA HORA ANT. (AUT) (%)                 float64
UMIDADE RELATIVA DO AR, H

### Verificando a média da temperatura

In [8]:
coluna_temperatura  = df['TEMPERATURA DO AR - BULBO SECO, HORARIA (°C)']

In [9]:
coluna_temperatura = pd.to_numeric(df['TEMPERATURA DO AR - BULBO SECO, HORARIA (°C)'], errors='coerce')

In [10]:
coluna_temperatura.mean()

np.float64(26.69953596287703)

##### A média da temperatura do ar - bulbo seco, na cidade de João Pessoa, Paraíba, foi de 26.7 °C. 

### Retirando Nulos

In [12]:
df['RADIACAO GLOBAL (Kj/m²)'].isnull()

0        True
1        True
2        True
3        True
4        True
        ...  
8755    False
8756    False
8757    False
8758     True
8759     True
Name: RADIACAO GLOBAL (Kj/m²), Length: 8760, dtype: bool

In [13]:
remover = df['RADIACAO GLOBAL (Kj/m²)'].dropna(axis=0, how='all')
remover

9           62
10       368,5
11        1112
12      2064,6
13      2229,4
         ...  
8753    2814,6
8754    2184,6
8755    1523,9
8756     303,1
8757      40,3
Name: RADIACAO GLOBAL (Kj/m²), Length: 4764, dtype: object

### Reduzir DataFrame

In [14]:
df_reduzido  = df[['Data', 'PRESSÃO ATMOSFERICA MAX.NA HORA ANT. (AUT) (mB)', 'TEMPERATURA MÁXIMA NA HORA ANT. (AUT) (°C)']]

In [15]:
df_reduzido = df_reduzido.sample(n=1000)

In [22]:
df_reduzido.shape

(1000, 3)

### Bônus 1: Normalizar coluna

In [29]:
coluna_pressao = pd.to_numeric(df['TEMPERATURA DO AR - BULBO SECO, HORARIA (°C)'], errors='coerce')

In [32]:
df['Coluna1_normalizada'] = (df['UMIDADE REL. MAX. NA HORA ANT. (AUT) (%)'] - df['UMIDADE REL. MAX. NA HORA ANT. (AUT) (%)'].min()) / (df['UMIDADE REL. MAX. NA HORA ANT. (AUT) (%)'].max() - df['UMIDADE REL. MAX. NA HORA ANT. (AUT) (%)'].min())